valhalla/thor/dijkstras.h

```diff
@@ -194,29 +234,186 @@ class Dijkstras {
    * @param from_transition Boolean indicating if this expansion is from a transition edge.
    * @param time_info Tracks time offset as the expansion progresses
    */
-  void ExpandForward(baldr::GraphReader& graphreader,
+  template <const ExpansionType expansion_direction,
+            // As multimodal isn't ported yet, but we're using the 3-value enum, check that
+            // the caller is using a supported expansion mode (this is a compile time check to
+            // catch this mistake early).  Once multimodal is supported, this can be removed.
+            class = typename std::enable_if<expansion_direction == ExpansionType::forward || expansion_direction == ExpansionType::reverse>::type>
+  void ExpandTemplate(baldr::GraphReader& graphreader,
                      const baldr::GraphId& node,
                      const sif::EdgeLabel& pred,
                      const uint32_t pred_idx,
+                     const baldr::DirectedEdge *opp_pred_edge,
```

Comment:
Templatizing this consolidates the signature, and requires callers in the forward case to supply something for `opp_pred_edge` - nullptr in most cases.  Does anyone have an issue with that general pattern?

Reply:
Works for me.